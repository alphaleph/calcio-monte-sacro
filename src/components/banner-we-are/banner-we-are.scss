@import '../../pages/globals.scss';

.is-transparent {
    opacity: 0 !important;
}

.we-are-container {
    padding-top: 1rem;
    border-radius: 1px;
}

.we-are-text {
    color: rgba(255, 246, 231, .6);;
    font-size: 2.5rem;
    margin-bottom: 1rem;
    opacity: 1 !important;
    transition: opacity 1s ease-in;
}

.we-are-name {
    color: rgba(252, 233, 95, .8);
    text-shadow:
        -1px -1px 0 $granite-grey-team,
        1px -1px 0 $granite-grey-team,
        -1px 1px 0 $granite-grey-team,
        1px 1px 0 $granite-grey-team;
    font-size: 4.5rem;
    opacity: 1 !important;
    transition: opacity 1s ease-in 1s;
}

.placeholder-pic {
    font-size: 15rem;
}