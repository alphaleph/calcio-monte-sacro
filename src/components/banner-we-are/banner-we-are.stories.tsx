import React from 'react';
import { storiesOf } from '@storybook/react';
import { BannerWeAre } from './banner-we-are';
import '../../pages/index.scss';

storiesOf('BannerWeAre', module).add('default', () => <BannerWeAre />);
