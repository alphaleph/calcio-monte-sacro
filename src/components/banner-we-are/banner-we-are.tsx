import React, { useState, useEffect } from 'react';
import './banner-we-are.scss';
import { useStaticQuery, graphql } from 'gatsby';
import Img from 'gatsby-image';
import { FluidImageData } from '../../types';

interface BannerWeAreOwnProps {}
interface BannerWeAreProps extends BannerWeAreOwnProps {
    bgImage: FluidImageData;
}
export const PureBannerWeAre = ({ bgImage }: BannerWeAreProps) => {
    const [isLoaded, setIsLoaded] = useState(false);
    useEffect(() => {
        setIsLoaded(true);
    }, []);

    //TODO: Implement Canvas BG, animate with PixiJS
    return (
        <section>
            <Img fluid={bgImage.fluid} className="banner-we-are-background" />
            <div className="banner-we-are-foreground">
                <h1>
                    <span
                        className={
                            isLoaded
                                ? 'banner-we-are-name is-family-secondary'
                                : 'banner-we-are-name is-transparent is-family-secondary'
                        }
                    >
                        Calcio Monte Sacro
                    </span>
                </h1>
            </div>
        </section>
    );
};

export const BannerWeAre = () => {
    const data = useStaticQuery(graphql`
        query {
            placeholderImage: file(
                relativePath: { eq: "mountaintop-sky-temp.png" }
            ) {
                childImageSharp {
                    fluid(maxWidth: 1024) {
                        base64
                        aspectRatio
                        src
                        srcSet
                        sizes
                        originalName
                    }
                }
            }
        }
    `);

    return <PureBannerWeAre bgImage={data.placeholderImage.childImageSharp} />;
};
