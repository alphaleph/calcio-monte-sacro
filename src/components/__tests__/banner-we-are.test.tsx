import React from 'react';
import { BannerWeAre } from '../banner-we-are/banner-we-are';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

describe('We Are Banner', () => {
    it('renders without crashing', () => {
        const { getByText } = render(<BannerWeAre />);
        expect(getByText('Calcio Monte Sacro')).toBeInTheDocument();
    });

    it('renders correctly', () => {
        const { getByText } = render(<BannerWeAre />);
        expect(getByText('Calcio Monte Sacro')).toBeVisible();
    });

    it('has not changed', () => {
        const { asFragment } = render(<BannerWeAre />);
        expect(asFragment()).toMatchSnapshot();
    });
});
