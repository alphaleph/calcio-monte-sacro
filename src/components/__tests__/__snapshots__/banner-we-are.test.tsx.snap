// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`We Are Banner has not changed 1`] = `
<DocumentFragment>
  <section
    class="hero has-background-night-sky is-bold"
  >
    <div
      class="hero-body"
    >
      <div
        class="we-are-container container has-text-centered"
      >
        <h1
          class="title"
        >
          <span
            class="we-are-text"
          >
            We are...
          </span>
          <br />
          <br />
          <span
            class="we-are-name is-family-secondary"
          >
            Calcio Monte Sacro
          </span>
        </h1>
      </div>
    </div>
    <p
      class="has-text-centered placeholder-pic"
    >
      ðŸ—»
    </p>
  </section>
</DocumentFragment>
`;
